web/src/app/util/SpeedDial.js

```diff
@@ -45,8 +45,9 @@ export default function CustomSpeedDial(props) {
             classes={{ staticTooltipLabel: classes.staticTooltipLabel }}
             aria-label={action.label}
             onClick={() => {
-              setOpen(false)
-              action.onClick()
+              if (!action.disabled) {
```

Comment:
Let's invert this to get rid of the double negative, and so the primary behavior isn't nested in a block.

We can also make this work a little better:
- If we call `stopPropagation`, it will prevent the SpeedDial from closing when clicking the disabled item
- We can add `TooltipClasses` to "disable" the tooltip text [info here](https://material-ui.com/api/speed-dial-action/#props)
- Add `FabProps={{disabled: action.disabled}}` to use the built-in disable styling for the button [from here](https://material-ui.com/api/fab/#props) instead of manually changing the icon color

// click handler
```js
onClick={e => {
  if (action.disabled) {
    e.stopPropagation()
    return
  }
  action.onClick()
}
```